iothub/service/src/Registry2/ConfigurationsClient.cs

```diff
@@ -0,0 +1,341 @@
+        /// <param name="forceUpdate">Forces the configuration object to be replaced even if it was updated since it was retrieved last time.</param>
+        /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
+        /// <returns>The Configuration object with updated ETags.</returns>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public async virtual Task<Configuration> UpdateAsync(Configuration configuration, bool forceUpdate, CancellationToken cancellationToken = default)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Updating configuration: {configuration?.Id} - Force update: {forceUpdate}", nameof(UpdateAsync));
+
+            try
+            {
+                if (string.IsNullOrWhiteSpace(configuration.ETag) && !forceUpdate)
+                {
+                    throw new ArgumentException(ApiResources.ETagNotSetWhileUpdatingConfiguration);
+                }
+
+                using HttpRequestMessage request = _httpRequestMessageFactory.CreateRequest(HttpMethod.Put, GetConfigurationRequestUri(configuration.Id), _credentialProvider, configuration);
+                HttpMessageHelper2.InsertEtag(request, configuration.ETag);
+                HttpResponseMessage response = await _httpClient.SendAsync(request, cancellationToken);
+                await HttpMessageHelper2.ValidateHttpResponseStatus(HttpStatusCode.OK, response);
+                return await HttpMessageHelper2.DeserializeResponse<Configuration>(response, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"{nameof(UpdateAsync)} threw an exception: {ex}", nameof(UpdateAsync));
+                throw;
+            }
+            finally
+            {
+                if (Logging.IsEnabled)
+                    Logging.Exit(this, $"Updating configuration: {configuration?.Id} - Force update: {forceUpdate}", nameof(UpdateAsync));
+            }
+        }
+
+        /// <summary>
+        /// Deletes a previously registered device from the system.
+        /// </summary>
+        /// <param name="configurationId">The id of the configurationId being deleted.</param>
+        /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public async virtual Task RemoveAsync(string configurationId, CancellationToken cancellationToken = default)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Removing configuration: {configurationId}", nameof(RemoveAsync));
+
+            try
+            {
+                if (string.IsNullOrWhiteSpace(configurationId))
```

Comment:
This if statement can be simplified to just call Argument.RequireNotNullOrEmpty

[example](https://github.com/Azure/azure-iot-sdk-csharp/blob/4164acb88e987cfbf92ed3b97c8fd14c17dc6a06/iothub/service/src/Registry2/DevicesClient.cs#L123)